tar/multitape/multitape_write.c

```diff
@@ -747,22 +752,26 @@ flushtape(TAPE_W * d, int isapart, int extrastats)
 	tmi.tindexlen = d->t.indexlen;
 
 	/*
-	 * Store archive metaindex.  Note that this must be done before the
-	 * archive metadata is stored, since it fills in fields in the archive
-	 * metadata concerning the index length and hash.
+	 * If not a dry run, store archive metaindex.  Note that this must be
+	 * done before the archive metadata is stored, since it fills in
+	 * fields in the archive metadata concerning the index length and
+	 * hash.
 	 */
-	if (multitape_metaindex_put(d->S, d->C, &tmi, &tmd, extrastats))
+	if ((d->dryrun == 0) && (multitape_metaindex_put(d->S, d->C, &tmi, &tmd, extrastats)))
```

Comment:
Doesn't this make the statistics from `tarsnap -c --dry-run --print-stats` wrong?  We need to create the metaindex and metadata in order to account for them in the statistics, even if we're not going to upload them.